Tests/DecodableTests.swift

```diff
@@ -30,242 +30,281 @@ class DecodableTestCase: BaseTestCase {
 
     // MARK: - Decodable API Success Tests
 
-    func testThatParseOnADecodableSucceeds() {
-        // Given
-        let data = loadJSONDataForFileNamed("PropertyTypesTest")
-
-        // When
+    func testThatDecodeObjectWithDecodableTypeSucceeds() {
         do {
+            // Given
```

Comment:
The purpose is to condense the test as much as possible to make it easier to code scan. If everything exists inside the `do`, we don't have to reason why some of it is inside and some of it is outside. It's something we've done across all our test suites and makes it easier to breeze through the tests from a consistency standpoint. 

Another reason is to make it easier to add additional `try` declarations to the same test if necessary. If any of the `try` calls throw, the test should just stop. No point in trying to continue. One big do-catch declaration is the most consistent way to handle all these cases. Hence the change.

There is certainly no functional reason for the change. It's purely cosmetic for code scanning, consistency and extensibility.